src/main/lrsql/spec/admin/status.clj

```diff
@@ -0,0 +1,114 @@
+(ns lrsql.spec.admin.status
+  (:require [clojure.spec.alpha :as s]
+            [lrsql.backend.protocol :as bp]
+            [lrsql.spec.common :as common]
+            [lrsql.util :as u]
+            [xapi-schema.spec :as xs]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Interface
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn admin-status-backend?
+  [bk]
+  (satisfies? bp/AdminStatusBackend bk))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Axioms
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/def ::count nat-int?)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Inputs
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/def ::status-query-name
+  #{"statement-count"
+    "actor-count"
+    "platform-frequency"
+    "last-statement-stored"
+    "timeline"})
+
+(s/def :lrsql.spec.admin.status.params/include
+  (s/nonconforming
+   (s/or :single   ::status-query-name
+         :multiple (s/every ::status-query-name))))
+(s/def :lrsql.spec.admin.status.params/timeline-unit
+  #{"year"
+    "month"
+    "day"
+    "hour"
+    "minute"
+    "second"})
+
+(defn- after-epoch?
+  [timestamp]
+  (-> timestamp
+      u/str->time
+      u/time->millis
+      nat-int?))
+
+(s/def :lrsql.spec.admin.status.params/timeline-since
+  (s/and ::xs/timestamp
+         after-epoch?))
+(s/def :lrsql.spec.admin.status.params/timeline-until
+  (s/and ::xs/timestamp
+         after-epoch?))
+
+
+(def get-status-params-spec
+  (s/keys :opt-un [:lrsql.spec.admin.status.params/include
+                   :lrsql.spec.admin.status.params/timeline-unit
+                   :lrsql.spec.admin.status.params/timeline-since
+                   :lrsql.spec.admin.status.params/timeline-until]))
```

Comment:
You might want to add restrictions on if the `timeline-unit` covers an interval greater than the interval between `timeline-since` and `timeline-util` (e.g. you choose year as your unit but your since-to-until interval only covers two months). Right now in the UI nothing shows up in the timeline in that case and it feels pretty weird.

Reply:
I'll look into this on the front-end. On the back end I think it's just GIGO that it doesn't make sense to spend the time to police.